cli/providers/deploy.go

```diff
@@ -0,0 +1,18 @@
+package providers
+
+import (
+	"fmt"
+
+	"github.com/pchico83/i2kit/cli/providers/aws"
+	"github.com/pchico83/i2kit/cli/schemas/environment"
+	"github.com/pchico83/i2kit/cli/schemas/service"
+)
+
+//Deploy deploys a given service in a given environment
+func Deploy(s *service.Service, e *environment.Environment) error {
+	if e.Provider == nil {
+		fmt.Println("Service dry-run deployed succesfully. Did you define an 'environment.yml' file?")
+		return nil
```

Comment:
Could we do it like before were this was generating the cloud formation template but not executing it?

Reply:
Also, I think we should leave the comment as 'Service {name} dry-run deployment was success' (or somethign like that). The question about environment.yaml makes it feel as it this could be an error.

Reply:
1 - It is not trivial, but doable. What would be the value added? I think we should have this just for demo/play purposes, we might have more providers in the feature. I would leave it like this, or maybe print the auto-generated compose file, and if necessary, add a `dry-run` option for every provider. The dry-run would be more powerful then, it would check credentials, quotas... calling the AWS Cloud Formation creation with `dry-run`.
2 - Ok

Reply:
I do like the idea of dry-run, as a way to validate that all the values are correct and available, before doing an actual deployment (since deploying costs money). 

I do like being able to do a 'demo' deployment where it fakes it and just returns some pretend logs. I'm not sure if the aws-level demo provider is where we want to make this available. What do you think?

Reply:
Agreed. Let's print the generated compose in this PR for the second case, and I will add a `dry-run` for aws in a another PR.